### ISSUE TITLE
Can this app co-exist with HA?

### ISSUE THREAD

**Issue by riccidan** [thirtythreeforty/neolink#202]
Hi there,

Great work on this application by the way, you seem to be the only person that has figured out a solution to this issue.....and apologies for my relatively noob question. 

I'd noticed someone else is also looking to deploy your software to bridge their ipcamera to Home Assistant, but would your software require a dedicated rpi? Or could it be run on the HA pi. 

My rpi4 is utilising little to no resources so far and this would be an ideal outcome, next question is of course, how do i go about installing it on the same pi?
![Screenshot 2021-08-27 065741](https://user-images.githubusercontent.com/69840406/131035062-9b3a5b3e-2319-46d5-b205-bd42ea3d826b.png)

**Comment by QuantumEntangledAndy** [comment]
I did a little reading too and it seems to be all geared towards writing an integration. These integrations are in python which makes it a bit of a snag for neolink to integrate with since it's in rust. One day I hope to have working c and python bindings in neolink and I might try and work up an integration then for my own homeassistant.

My homeassistant is not the hasos but it's a standard rasbian I installed ha on. So I've got a bit more control of it making it easier install neolink and other fun stuff on it too.

Sorry it seems difficult to do for the haos version :(

**Comment by riccidan** [comment]
No apology necessary Mate, hats off to you for writing this, I read your blog that shared your experience and was truly impressed. I'm now thinking of purchasing a used pi and running Neolink on its own, its a fine line though....if I spend too much on hardware I could just as easily migrate to a cam with RTSP. May reach out to you for help with that if I find one. :)

**Comment by QuantumEntangledAndy** [comment]
The blog post is @thirtythreeforty's which I agree is a nice read.

You could just use your current pi but change it from haos to Debian and install ha on that. Just need to back up and restore your haos stuff first. Still a new pi might be easier if you don't mind the expense. Your call.

**Comment by mttyd** [comment]
Curious, would it be possible to run your app on a standalone box and then use this library https://github.com/fwestenberg/reolink_dev to read it from HA?  Also any chance your library will work on an argus 2?

**Comment by QuantumEntangledAndy** [comment]
I don't think that integration will work in the way your suggesting (if I understand what your saying). But if you run it on a standalone box you can just add it as a generic rtsp camera instead of using that reolink integration. 

We are currently working on Argus 2 support, it should work once #199 is merged.

**Comment by dm82m** [comment]
@QuantumEntangledAndy hey andrew, I tried to create a HAOS Add-on for Neolink but I am somehow stuck

**config.yaml**
```
name: "Neolink"
description: "Will run Neolink directly as Add-on in HAOS!"
version: "1.0.0"
slug: "neolink"
init: false
arch:
  - aarch64
startup: before
ports:
  8554/tcp: 8554
```

**Dockerfile**
```
ARG BUILD_FROM
FROM $BUILD_FROM

RUN apk add \
  -X http://dl-cdn.alpinelinux.org/alpine/edge/main \
  -X http://dl-cdn.alpinelinux.org/alpine/edge/testing \ 
  wget \
  gstreamer \
  gst-plugins-base \
  gst-plugins-good \
  gst-plugins-bad \
  gst-plugins-ugly \
  gst-rtsp-server

RUN wget https://nightly.link/thirtythreeforty/neolink/workflows/build/master/release-arm64-buster.zip

RUN unzip release-arm64-buster.zip

RUN cp neolink /usr/local/bin/

RUN chmod +x /usr/local/bin/neolink

# Copy data for add-on
COPY run.sh /
RUN chmod a+x /run.sh

CMD [ "/run.sh" ]
```

**run.sh**
```
#!/usr/bin/with-contenv bashio

echo "Starting neolink ..."

neolink rtsp --config my_config.toml
```

But this ends with:
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
Starting neolink ...
**/run.sh: line 5: /usr/local/bin/neolink: No such file or directory**
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

Any idea what I am doing wrong? I know that I am not taking care of the config file atm. But this will then be the next step after I got neolink running.

Thanks for your help!

**Comment by dm82m** [comment]
I created an Add-on for HAOS that runs neolink. Cannot confirm 100% that its working due to lack of according Reolink cam. Cam is on its way to me. Let me know if you are interested in the Add-on. I will maybe donate that to the HA community.

**Comment by QuantumEntangledAndy** [comment]
@dm82m Did you get your issue with file not found sorted. If you can ensure that the file exists (which should be based on what I saw) then it is likely the interpreter of your os is in a different location then where it was when it was built.

Your choices are build it yourself or use patch elf to change the expected location of the interpreter in the neolink binary or use the same os it was built with (Debian).

**Comment by dm82m** [comment]
Yes it’s solved. I switched from alpine container to Debian. Add-on starts neolink. Can’t Test if everything works correctly cause i have no reolink at Hand.

https://github.com/dm82m/hassio-addons
